Actions/ReadPowerPlatformSettings/ReadDeploymentAndAuthSettings.ps1

```diff
@@ -0,0 +1,54 @@
+param(
+    [Parameter(Mandatory = $true)]
+    [string] $deploymentEnvironmentsJson,
+    [Parameter(Mandatory = $true)]
+    [string] $environmentName
+)
+
+$deploymentEnvironments = $deploymentEnvironmentsJson | ConvertFrom-Json
+$deploymentSettings = $deploymentEnvironments."$environmentName"
+$envName = $environmentName.Split(' ')[0]
+$secrets = $env:Secrets | ConvertFrom-Json
+
+foreach($property in 'ppEnvironmentUrl','companyId','environmentName') {
+    if ($deploymentSettings."$property") {
+        Write-Host "Setting $property"
+        Add-Content -Encoding utf8 -Path $env:GITHUB_OUTPUT -Value "$property=$($deploymentSettings."$property")"
+    }
+    else {
+        Write-Host "::ERROR::DeployTo$envName setting must contain '$property' property"
+        exit 1
+    }
+}
+
+$authContext = $null
+foreach($secretName in "$($envName)-AuthContext","$($envName)_AuthContext","AuthContext") {
+    if ($secrets."$secretName") {
+        Write-Host "Setting authentication context from secret $secretName"
+        $authContext = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secrets."$secretName")) | ConvertFrom-Json
+        'ppTenantId','ppApplicationId','ppClientSecret','ppUserName','ppPassword' | ForEach-Object {
+            if ($authContext.PSObject.Properties.Name -eq $_) {
+                Write-Host "Setting $_"
+                Add-Content -Encoding utf8 -Path $env:GITHUB_OUTPUT -Value "$_=$($authContext."$_")"
+                Set-Variable -Name $_ -Value $authContext."$_"
+            }
+            else {
+                Add-Content -Encoding utf8 -Path $env:GITHUB_OUTPUT -Value "$_="
+                Set-Variable -Name $_ -Value ""
+            }
+        }
+
+        if ($ppApplicationId -and $ppClientSecret -and $ppTenantId) {
+            Write-Host "Authenticating with application ID and client secret"
+        }
+        elseif ($ppUserName -and $ppPassword -and $ppTenantId) {
```

Comment:
PPTenantId is not needed for username password auth - it should be fine to just remove the check. 

Note: The write-host comment on line 48 matches the "no-tenantId" check